platforms/core-runtime/internal-instrumentation-api/src/main/java/org/gradle/internal/instrumentation/api/types/BytecodeInterceptorRequest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.instrumentation.api.types;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * Request for interceptors. Currently, we support filter just for interception types, but in the feature we could also filter per version etc.
+ *
+ * Implemented as a enum, so it's easier to generate bytecode.
+ */
+public enum BytecodeInterceptorRequest {
```

Comment:
🤔 Would it make sense to find a name that would better reflect that this entity does not contain any real "payload" of the request? Looking at the word "request" at the use sites made me think about more complex data structure being involved. Maybe something about interception context filter?

Reply:
Yes, I can rename that. Maybe we can go with BytecodeInterceptorFilter. And if we will add more context later maybe we can rename it